ddtrace/tracer/tracer.go

```diff
@@ -173,7 +173,7 @@ func newTracer(opts ...StartOption) *tracer {
 	t := &tracer{
 		config:           c,
 		payload:          newPayload(),
-		flushChan:        make(chan struct{}),
+		flushChan:        make(chan struct{}, 1),
```

Comment:
Instead of fine tuning the flush channel, let's aim to get rid of it completely, as discussed :)